content/en/blog/2023/creating-k8s-cluster-with-runtime-observability.md

```diff
@@ -0,0 +1,472 @@
+{{end}}
+{{end}}
+{{end}}
+
+{{range $k, $v := .PrivateRegistryConfig.Configs }}
+{{ if $v.Auth }}
+[plugins."io.containerd.grpc.v1.cri".registry.configs."{{$k}}".auth]
+  {{ if $v.Auth.Username }}username = {{ printf "%q" $v.Auth.Username }}{{end}}
+  {{ if $v.Auth.Password }}password = {{ printf "%q" $v.Auth.Password }}{{end}}
+  {{ if $v.Auth.Auth }}auth = {{ printf "%q" $v.Auth.Auth }}{{end}}
+  {{ if $v.Auth.IdentityToken }}identitytoken = {{ printf "%q" $v.Auth.IdentityToken }}{{end}}
+{{end}}
+{{ if $v.TLS }}
+[plugins."io.containerd.grpc.v1.cri".registry.configs."{{$k}}".tls]
+  {{ if $v.TLS.CAFile }}ca_file = "{{ $v.TLS.CAFile }}"{{end}}
+  {{ if $v.TLS.CertFile }}cert_file = "{{ $v.TLS.CertFile }}"{{end}}
+  {{ if $v.TLS.KeyFile }}key_file = "{{ $v.TLS.KeyFile }}"{{end}}
+  {{ if $v.TLS.InsecureSkipVerify }}insecure_skip_verify = true{{end}}
+{{end}}
+{{end}}
+{{end}}
+
+{{range $k, $v := .ExtraRuntimes}}
+[plugins."io.containerd.grpc.v1.cri".containerd.runtimes."{{$k}}"]
+  runtime_type = "{{$v.RuntimeType}}"
+[plugins."io.containerd.grpc.v1.cri".containerd.runtimes."{{$k}}".options]
+  BinaryName = "{{$v.BinaryName}}"
+{{end}}
+
+[plugins."io.containerd.tracing.processor.v1.otlp"]
+  endpoint = "host.k3d.internal:4317"
+  protocol = "grpc"
+  insecure = true
+
+[plugins."io.containerd.internal.v1.tracing"]
+  sampling_ratio = 1.0
+  service_name = "containerd"
+```
+
+After creating these files, open a terminal inside the `kubetracing` folder and
+run `k3d` to create a cluster. Before running this command, replace the
+`[CURRENT_PATH]` placeholder for the entire path of the `kubetracing` folder.
+You can retrieve it by running the `echo $PWD` command in the terminal in that
+folder.
+
+```toml
+k3d cluster create tracingcluster \
+  --image=rancher/k3s:v1.27.1-k3s1 \
+  --volume '[CURRENT_PATH]/config.toml.tmpl:/var/lib/rancher/k3s/agent/etc/containerd/config.toml.tmpl@server:*' \
+  --volume '[CURRENT_PATH]/config:/etc/kube-tracing@server:*' \
+  --k3s-arg '--kube-apiserver-arg=tracing-config-file=/etc/kube-tracing/apiserver-tracing.yaml@server:*' \
+  --k3s-arg '--kube-apiserver-arg=feature-gates=APIServerTracing=true@server:*' \
+  --k3s-arg '--kubelet-arg=config=/etc/kube-tracing/kubelet-tracing.yaml@server:*'
+```
+
+This command will create a Kubernetes cluster with version `v1.17.1`, and set up
+in three docker containers on your machine. If you run the command
+`kubectl cluster-info` now, you will see this output:
+
+```bash
+Kubernetes control plane is running at https://0.0.0.0:60503
+CoreDNS is running at https://0.0.0.0:60503/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy
+Metrics-server is running at https://0.0.0.0:60503/api/v1/namespaces/kube-system/services/https:metrics-server:https/proxy
+```
+
+Going back to the logs of the observability environment, you should see some
+spans of internal Kubernetes operations being published in OpenTelemetry
+Collector, like this:
+
+```bash
```

Comment:
This isn't `bash` code. Consider dropping the language specifier?

```suggestion
```
```

Reply:
It makes sense, I've dropped the language specifier.